api/hasura/actions/create_nominee.ts

```diff
@@ -0,0 +1,88 @@
+import assert from 'assert';
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { z } from 'zod';
+
+import {
+  insertNominee,
+  getUserFromProfileIdWithCircle,
+  getUserFromAddress,
+  getNomineeFromAddress,
+} from '../../../api-lib/createNomineeMutations';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+import { GraphQLError } from '../../../src/lib/gql/zeusHasuraAdmin';
+import {
+  createNomineeSchemaInput,
+  composeHasuraActionRequestBody,
+} from '../../../src/lib/zod';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  try {
+    const {
+      input: { object: input },
+      session_variables: sessionVariables,
+    } = composeHasuraActionRequestBody(createNomineeSchemaInput).parse(
+      req.body
+    );
+    const { circle_id, address, name, description } = input;
+
+    if (sessionVariables.hasuraRole !== 'admin') {
```

Comment:
Instead of putting all the logic in an if block like this, it would be easier to follow if you instead terminated early if the condition was not met. 

e.g. 

```ts
if (sessionVariables.hasuraRole === 'admin')
  return res.status(...);

// rest of the logic
```

Otherwise its difficult to figure out the code path